proposals/0020-distributed-actor-cluster.md

```diff
@@ -0,0 +1,172 @@
+
+```swift
+typealias DefaultDistributedActorSystem = ClusterSystem
+```
+
+Which avoids having to repeat the `ActorSystem` typealias in every distributed actor in a module.
+
+The package being proposed here provides the implementation of the `ClusterSystem`.
+
+## Proposed solution
+
+The `DistributedCluster` package includes the `ClusterSystem` type which is the central piece of the library. Once created, it binds to a host/port pair and begins listening for incoming connections:
+
+```swift
+@main
+struct Main {
+    static func main() async throws {
+        let system = await ClusterSystem("FirstSystem") { settings in
+            settings.endpoint.host = "127.0.0.1"
+            settings.endpoint.port = 7337
+        }
+        
+        try await system.terminated
+    }
+}
+```
+
+Clusters are formed by “joining” other nodes. So the above created node has bound on port 7337, and if we had another node running on the same host but on port 8228 we could join in programmatically like this:
+
+```swift
+let other = Cluster.Endpoint(host: "127.0.0.1", port: 8228)
+system.cluster.join(endpoint: other)
+```
+
+The `join(endpoint:)` method returns immediately, however it is possible to `try await joined(endpoint:within:)` to suspend until the other node has been joined (or we failed doing so within the timeout).
+
+Alternatively we can configure an actor system to automatically discover and join nodes by configuring a [`swift-service-discovery` node-based discovery mechanism](https://apple.github.io/swift-distributed-actors/1.0.0-beta.2/documentation/distributedactors/clustering#Automatic-Node-Discovery), like this:
+
+```swift
+import ServiceDiscovery
+import K8sServiceDiscovery // See: [tuplestream/swift-k8s-service-discovery](https://github.com/tuplestream/swift-k8s-service-discovery)
+import DistributedCluster
+
+ClusterSystem("DiscoverNodes") { settings in
+    let discovery = K8sServiceDiscovery() 
+    let target = K8sObject(labelSelector: ["name": "actor-cluster"], namespace: "actor-cluster")
+    
+    settings.discovery = ServiceDiscoverySettings(discovery, service: target)
+}
+```
+
+Once a cluster has been formed, distributed actors use the [Receptionist](https://apple.github.io/swift-distributed-actors/1.0.0-beta.2/documentation/distributedactors/receptionist) to discover check-in and discover distributed actors from other nodes. For example, we can write a distributed worker actor like this:
+
+```swift
+distributed actor Worker {
+    typealias ActorSystem = ClusterSystem
+    
+    init(actorSystem: ActorSystem) async {
+        self.actorSystem = actorSystem
+        await system.receptionist.checkIn(worker, with: .workers) 
```

Comment:
where is `system` from here - is it a singleton?

Reply:
typo, thanks!

It is supposed to be `actorSystem`; my muscle memory won here hehe... I often call it system in the lib

```suggestion
        await actorSystem.receptionist.checkIn(worker, with: .workers) 
```